source/grid/tria_accessor.cc

```diff
@@ -1496,7 +1496,7 @@ TriaAccessor<structdim, dim, spacedim>::set_bounding_object_indices(
   const ArrayView<int> bounding_object_index_ref =
     this->objects().get_bounding_object_indices(this->present_index);
 
-  AssertDimension(bounding_object_index_ref.size(), new_indices.size());
+  // AssertDimension(bounding_object_index_ref.size(), new_indices.size());
```

Comment:
why do you need to remove this?

Reply:
I have replaced the assert now by an `AssertIndexRange`. The problem here `riaObjects::get_bounding_object_indices()` returns the maximal possible array size (working for simplices, ..., and for hypercubes).